### PR TITLE
adding swift wrapper for DiagnosticDataViewer

### PR DESCRIPTION
**Pull request by absaroj**

### CODE REVIEW

**Review by eduardo-camacho — APPROVED** [review]
Please get review from Lalit and Sid as well

**Review by lalitb — APPROVED** [review]
LGTM. Thanks. Is there any plan to add basic tests in [main.swift](https://github.com/microsoft/cpp_client_telemetry/blob/main/wrappers/swift/main.swift) for basic tests on this wrapper ?

File: wrappers/swift/DiagnosticDataViewer.swift
Comment (mannie) [inline wrappers/swift/DiagnosticDataViewer.swift:82]: Something like `onDisableNotification(run:)` would yield more readable Swift code at the usage site.

Comment (absaroj) [reply wrappers/swift/DiagnosticDataViewer.swift:82]: Thanks for the suggestion. Updated the api.

File: wrappers/swift/DiagnosticDataViewer.swift
Comment (mannie) [inline wrappers/swift/DiagnosticDataViewer.swift:7]: Do we expect subclasses of this class? If not, we should mark this as `final` to prevent unexpected behaviors if someone chooses to subclass.

File: wrappers/swift/DiagnosticDataViewer.swift
Comment (mannie) [inline wrappers/swift/DiagnosticDataViewer.swift:17]: Can we rename this to something like `instantiateViewer(machineIdentifier:)`, please?

File: wrappers/swift/DiagnosticDataViewer.swift
Comment (mannie) [inline wrappers/swift/DiagnosticDataViewer.swift:61]: The Swifty way of writing getters and setters would be as computed variables; it would be nice for us to leverage that pattern here and in the other PRs :)

Comment (absaroj) [reply wrappers/swift/DiagnosticDataViewer.swift:61]: can you please explain this one? as i see, only viewEnabled can be extracted to computed properties and rest of the viewer methods would need to be as methods.

Comment (mannie) [reply wrappers/swift/DiagnosticDataViewer.swift:61]: My mistake. I meant to put this one function lower.

```
    public static var viewerEnabled: Bool {
         ODWDiagnosticDataViewer.viewerEnabled()
     }

     public static var currentEndpoint: String {
         return ODWDiagnosticDataViewer.currentEndpoint()
     }
```

Comment (absaroj) [reply wrappers/swift/DiagnosticDataViewer.swift:61]: I see, but there are other related methods with these properties which can't be combined into setter computed properties, such as disableViewer.
Converting the just one part of the property into computed ones does not seem correct to me.

Comment (mannie) [reply wrappers/swift/DiagnosticDataViewer.swift:61]: `disableViewer` is a function performing a task, as opposed to a variable holding state. Variables, whether computed or stored, tend to communicate the storage of state in Swift, which functions communicate the performance of an action.

**Review by mannie — APPROVED** [review]

**Review by eduardo-camacho — APPROVED** [review]

**Comment by lalitb** [comment]
Merging to unblock. The iOS build is stuck not related to this PR.

**Pull request merged**
